libimage/layer_tree.go

```diff
@@ -91,17 +91,27 @@ func (l *layerNode) repoTags() ([]string, error) {
 
 // layerTree extracts a layerTree from the layers in the local storage and
 // relates them to the specified images.
-func (r *Runtime) layerTree(ctx context.Context, images []*Image) (*layerTree, error) {
-	layers, err := r.store.Layers()
-	if err != nil {
-		return nil, err
-	}
-
-	if images == nil {
-		images, err = r.ListImages(ctx, nil, nil)
+// layerTree require both parameters not empty or nil, to avoid inconsistences
+// between layers and images. If both parameters are nil then layers and images will be obtained from local storage.
+// You can obtain consistent layer and images with storage.MultiList.
+func (r *Runtime) layerTree(images []*Image, layers []storage.Layer) (*layerTree, error) {
+	switch is_consistent := (images != nil || layers == nil) && (images == nil || layers != nil); is_consistent {
```

Comment:
(This way of framing the condition seems to be correct but it also reads extremely non-obvious to me. How does one arrive at this, without explicitly relying on De Morgan’s laws?(